free-programming-books.md

```diff
@@ -1793,52 +1785,49 @@ For resources on Angular, Backbone, D3, Dojo, Ember, Express, jQuery, Knockout,
 * [Code Like a Pythonista: Idiomatic Python](http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html)
 * [CodeCademy Python](https://www.codecademy.com/learn/python)
 * [Composing Programs](http://composingprograms.com) (3.x)
-* [Data Structures and Algorithms in Python](http://www.brpreiss.com/books/opus7/html/book.html) - B. R. Preiss
+* [Data Structures and Algorithms in Python](http://multimedia.ucc.ie/Public/training/cycle1/algorithms-in-python.pdf) - B. R. Preiss (PDF)
```

Comment:
Yes,it looks like another authers of the sanme title. I found the history of Book.
https://web.archive.org/web/20161016153130/http://www.brpreiss.com/books/opus7/html/book.html